packages/sdk/src/AbstractConfigManager.ts

```diff
@@ -199,6 +199,26 @@ export abstract class AbstractConfigManager {
         };
     }
 
+    public async promptForDeployDirectory(defaultServerPath: string): Promise<string> {
+        const result = await this.showMenu({
+            items: [{ label: defaultServerPath }, { label: "$(plus) Add New Deploy Directory" }],
+            placeholder: `Select a deploy directory for the Zowe SSH server`,
+        });
+
+        if (!result) return;
+
+        if (result === "$(plus) Add New Deploy Directory") {
+            const input = await this.showInputBox({
+                title: "Enter custom deploy directory for the Zowe SSH server",
+                placeHolder: "E.g. " + defaultServerPath,
+            });
+
+            return input?.trim() || defaultServerPath;
```

Comment:
We should not accept a single space as a deploy directory. I'd recommend adding input validation to prevent that path from being valid input (such as `validateInput` for passing to VS Code input box API)